ultralytics/utils/checks.py

```diff
@@ -782,8 +783,37 @@ def cuda_is_available() -> bool:
     return cuda_device_count() > 0
 
 
+def check_soc(family: str = "rknn", node: str = "/proc/device-tree/compatible"):
```

Comment:
@lakshanthad All set. I need your help in testing this.

Reply:
@lakshanthad Could you please verify if this correctly detects that you are using a Rockchip chip?

Reply:
@ambitious-octopus Here are outputs from both RK3588 and RK3566
![radxa@rock-5b~$ cat procdevice-treecompatible](https://github.com/user-attachments/assets/ee2da0d4-35dc-4524-9262-0e7ac0577156)
![CleanShot 2025-01-16 at 06 48 35@2x](https://github.com/user-attachments/assets/f7c28f00-1983-4f6a-b850-3e429fbe26c4)